Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java

```diff
@@ -284,20 +287,28 @@ public INetwork getNetwork(String networkId) throws QuadrigaStorageException {
     public List<INetwork> getNetworkList(IUser user) throws QuadrigaStorageException {
         return networkmapper.getListOfNetworksForUser(user);
     }
-    
+
     /**
-    * 
-    * {@inheritDoc}
-    * 
-    * This implementation uses the hibernate for dataaccess from the database
-    */
+     * 
+     * {@inheritDoc}
+     * 
+     * This implementation uses the hibernate for dataaccess from the database
+     */
     @Override
     @Transactional
     public List<INetwork> getApprovedNetworkList() throws QuadrigaStorageException {
         return networkmapper.getListOfApprovedNetworks();
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getNetworkWithPopularTerms() throws QStoreStorageException {
+        return qStoreConnector.executeQuery(env.getProperty("allNetworks"), RELATION_EVENT);
```

Comment:
I would put the query part on the qstoreconnector level. This class should be as oblivious as possible regarding specific storage imlementation.